chore: refactor collector decoration (#133)
solarwinds/solarwinds-otel-collector-contrib: processor/solarwindsprocessor/README.md

SolarWinds processor adjusts OTel signals (metrics, logs, traces) to improve SolarWinds Observability experience.

## Features

> - Collector attributes decoration: configures collector-specific attributes and SolarWinds extension integration

Reviewer: That's factical technical description. But from feature perspective, I would like to have here what feature it activates or what brings you when the feature is activated.

Try similar to host attributes section.

- Host attributes decoration: determines and applies host identifiers and related attributes for proper entity creation in SolarWinds Observability SaaS platform
- Resource attributes: adds configured attributes to all signals
- Signal size monitoring: logs a warning when signals exceed configured size

## Configuration

### Minimal Configuration
```yaml
processors:
  solarwinds:
    collector_attributes_decoration:
      extension: solarwinds/sample
```

### Full Configuration
```yaml
processors:
  solarwinds:
    max_size_mib: 3
    collector_attributes_decoration:
      enabled: true
      extension: solarwinds/sample
    host_attributes_decoration:
      enabled: true
      on_prem_override_id: "custom-host-id"
    resource:
      custom.attribute: "value"
```

## Collector Attributes Decoration

> Processor provides collector-specific decorations and manages SolarWinds extension integration when enabled.

Reviewer: Extension is needed there for collector-specific decoration. It manages it because it is needed - this is probably not exact reason why to have it here noted.
```suggestion
Processor provides collector-specific decorations leveraging of configured `solarwinds` extension.
```

Reviewer: _In addition, this feature is enabled by default._

Let's emphasize this fact.

```yaml
collector_attributes_decoration:
  enabled: true
  extension: solarwinds/sample
```

Where:
- `enabled`: Enables collector attributes decoration (enabled by default)

> - `extension`: Identifies a SolarWinds Extension to use for obtaining required configuration

Reviewer: We need exact name configured...

```suggestion
- `extension`: Identifies a SolarWinds Extension by its full name for obtaining required configuration
```

### Extension Dependency

> When the `collector_attributes_decoration` is enabled (which it is by default), it requires `solarwinds` extension to be configured.

Reviewer: If you note default setup at the beginning, we can omit it here.

The processor automatically adds:
- `sw.otelcol.collector.name`: configured collector name
- `sw.otelcol.collector.entity_creation`: (optional) when entity inference is enabled in the extension, the SolarWinds OTel Collector entity will be inferred in SolarWinds Observability.

> > [!WARNING]

Reviewer: Too late for this warning. 😄  Can we move it up and incorporate as mentioned above?
